src/main/java/com/google/sps/servlets/addSchoolInfo.java

```diff
@@ -0,0 +1,179 @@
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Filter schoolFilter = new FilterPredicate("school-name", FilterOperator.EQUAL, schoolName);
+    Query schoolQ = new Query("School").setFilter(schoolFilter);
+    List<Entity> schoolQuery =
+        datastore.prepare(schoolQ).asList(FetchOptions.Builder.withDefaults());
+
+    if (schoolQuery.size() == 0) {
+      Entity newSchool = new Entity("School");
+      newSchool.setProperty("school-name", schoolName);
+      newSchoolKey = newSchool.getKey();
+      datastore.put(newSchool);
+    } else {
+      Entity existingSchool = schoolQuery.get(0);
+      existingSchoolKey = existingSchool.getKey();
+      newSchoolKey = null;
+    }
+
+    if (newSchoolKey != null) {
+      Entity newCourse = new Entity("Course", newSchoolKey);
+      newCourse.setProperty("course-name", courseName);
+      newCourse.setProperty("Units", units);
+      newCourseKey = newCourse.getKey();
+      datastore.put(newCourse);
+
+      Entity newProfessor;
+      newProfessor = new Entity("Professor", newSchoolKey);
+      newProfessor.setProperty("professor-name", profName);
+      newProfessorKey = newProfessor.getKey();
+      datastore.put(newProfessor);
+
+      Entity newTerm = new Entity("Term", newCourseKey);
+      newTerm.setProperty("term", term);
+      newTerm.setProperty("Professor", newProfessorKey);
+      datastore.put(newTerm);
+
+    } else {
+      Filter courseFilter = new FilterPredicate("course-name", FilterOperator.EQUAL, courseName);
+      Query courseQ = new Query("Course").setAncestor(existingSchoolKey).setFilter(courseFilter);
+      List<Entity> courseQuery =
+          datastore.prepare(courseQ).asList(FetchOptions.Builder.withDefaults());
+
+      // existing school --> new course
+      if (courseQuery.size() == 0) {
+        Entity newCourse;
+        if (existingSchoolKey != null) {
+          newCourse = new Entity("Course", existingSchoolKey);
+        } else {
+          newCourse = new Entity("Course", newSchoolKey);
+        }
+
+        newCourse.setProperty("course-name", courseName);
+        newCourse.setProperty("Units", units);
+        datastore.put(newCourse);
+
+        Entity newTerm = new Entity("Term", newCourseKey);
+        newTerm.setProperty("term", term);
+
+        Filter professorFilter =
+            new FilterPredicate("professor-name", FilterOperator.EQUAL, profName);
+        Query professorQ =
+            new Query("Professor").setAncestor(existingSchoolKey).setFilter(professorFilter);
+        List<Entity> professorQuery =
+            datastore.prepare(professorQ).asList(FetchOptions.Builder.withDefaults());
+
+        // existing school --> new course --> new professor
+        if (professorQuery.size() == 0) {
+          System.out.println("OLD SCHOOL NEW COURSE NEW PROFESSOR");
+          Entity newProfessor;
+          newProfessor = new Entity("Professor", existingSchoolKey);
```

Comment:
Why is this not Entity newProfessor = newEntity("Professor", existingSchoolKey)? I might be missing something but just wondering